basic-search-java/src/test/java/com/mydomain/example/SystemTestExample.java

```diff
@@ -0,0 +1,151 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+/**
+ * System tests - demonstrates:
+ * <ol>
+ *     <li>How to set up the endpoint to the test instance</li>
+ *     <li>Feed to the test instance</li>
+ *     <li>Query the test instance</li>
+ *     <li>Visit the test instance</li>
+ * </ol>
+ */
+public class SystemTestExample {
+
+    private String getTestEndpoint() {
+        if (false) {   // something here when running in pipeline
+            return "";
+        }
+        return "http://endpoint:10000/";
+    }
+
+
+    //@BeforeEach
+    public void init() throws Exception {
+        removeAllDocuments("document/v1/music/music/docid/");
+    }
+
+
+    //@Test  // Another annotation here so it will be run in pipeline
+    public void simpleGETSearchTest() throws Exception {
+
+        assertEquals(0, getSearchResults(searchDocumentsGET(getTestEndpoint()
+                + "/search/?yql="
+                + encode("select * from sources * where sddocname contains \"music\";", StandardCharsets.UTF_8))),
+                "Expected 0 documents before feeding");
+
+        postDocument(getTestEndpoint() + "document/v1/music/music/docid/1", "/music-data-1.json");
+        postDocument(getTestEndpoint() + "document/v1/music/music/docid/2", "/music-data-2.json");
+        assertEquals(2, getAllDocumentIds("document/v1/music/music/docid/").size(), "Expected 2 documents after feeding");
+
+        String searchResult = searchDocumentsGET(getTestEndpoint()
+                + "/search/?yql="
+                + encode("select * from sources * where title contains \"bad\";", StandardCharsets.UTF_8));
+        prettyPrint(searchResult);
+        assertEquals(2, getSearchResults(searchResult), "Find all docs!");
+
+        searchResult = searchDocumentsGET(getTestEndpoint()
+                + "/search/?yql="
+                + encode("select * from sources * where artist contains \"michael\";", StandardCharsets.UTF_8));
+        prettyPrint(searchResult);
+        assertEquals(1, getSearchResults(searchResult), "Find Michael Jackson!");
+    }
+
+
+    //@Test
+    public void simpleVisitTest() throws Exception {
+        postDocument(getTestEndpoint() + "document/v1/music/music/docid/1", "/music-data-1.json");
+        postDocument(getTestEndpoint() + "document/v1/music/music/docid/2", "/music-data-2.json");
+        assertEquals(2, getAllDocumentIds("document/v1/music/music/docid/").size(), "Expected 2 documents after feeding");
+
+        String visitResult = visitDocuments(getTestEndpoint() + "document/v1/music/music/docid/");
+        prettyPrint(visitResult);
+    }
+
+
+    public void removeAllDocuments(String path)  throws Exception {
+        HashSet<String> ids = getAllDocumentIds(path);
+        for (String id : ids) {
+            removeDocument(getTestEndpoint() + path + id.split(":")[4]);  // id:music:music::2
+        }
+    }
+
+    public HashSet<String> getAllDocumentIds(String path) throws Exception {
```

Comment:
Adding some comments for later dicussions:

This is something I think we'll provide in the library, and perhaps also as a building block for, e.g., delete all, like you do. 
It's definitely code that should be visible, and it very nicely demonstrates many aspects of visiting through document/v1.